proto/spacemesh/types.proto

```diff
@@ -66,13 +58,11 @@ message TransactionReceipt {
     TransactionId id = 1;           // the source transaction
     // The results of STF transaction processing
     enum TransactionResult {
```

Comment:
While I agree with you that this enum represents the output/result of the STF, and therefore must have a well-defined value, I disagree with removing the `UNDEFINED` value here simply because the value of a newly-instantiated data structure should not default to, e.g., `EXECUTED`. That could lead to hard-to-detect bugs. It should default to `UNDEFINED` at instantiation time and need to be subsequently set.

Reply:
I just can't imagine a case where the value of this enum will be undefined in its container message. It is not a valid value for this enum - when the node creates a receipt this enum must have one of the defined values. Adding another unused value actually can introduce bugs where the code that prepares a receipt forgets to set it. Requiring it to be explicit is a good thing in my opinion.

Reply:
but if you feel strongly about it then I can revert.

Reply:
> Requiring it to be explicit is a good thing in my opinion

I think we're in agreement here aren't we? My point is exactly this: that it should have to be explicitly set. Its default value should be an error state.

> It is not a valid value for this enum - when the node creates a receipt this enum must have one of the defined values

Exactly, I agree that it's not a valid value - which makes it easier to detect errors! If it ever has this value, then there's a bug in the system somewhere.

Reply:
> Adding another unused value actually can introduce bugs where the code that prepares a receipt forgets to set it

I actually think it makes this case less likely. If a receipt forgets to set it, and it defaults to a _valid_ value, that's much harder to spot than its defaulting to an invalid value

Reply:
I added it back for now.